src/NetworkEntities.js

```diff
@@ -59,13 +59,17 @@ class NetworkEntities {
     entity.firstUpdateData = entityData;
   }
 
-  updateEntity(client, dataType, entityData) {
+  updateEntity(client, dataType, entityData, source) {
     var networkId = entityData.networkId;
 
     if (this.hasEntity(networkId)) {
       this.entities[networkId].components.networked.networkUpdate(entityData);
     } else if (entityData.isFirstSync) {
-      this.receiveFirstUpdateFromEntity(entityData);
+      if (NAF.options.firstSyncSource && source !== NAF.options.firstSyncSource) {
+        NAF.log.write('Ignoring first sync from disallowed source', source);
```

Comment:
should we take any other action here in addition to ignoring the message? given that this seems to be only possible in the case of a malicious user, perhaps we should just start ignoring all subsequent messages from them?

Reply:
might want to add to the test suite too

Reply:
Good question about further actions. I think I'd like to get this out in the wild first before we start taking any more automatic actions. Even then, we may still want the action to be a manual kick, instead of an automatic global block.
Not going to tackle it with this push anyway.